cuda_core/cuda/core/experimental/_program.py

```diff
@@ -3,10 +3,353 @@
 # SPDX-License-Identifier: LicenseRef-NVIDIA-SOFTWARE-LICENSE
 
 import weakref
+from dataclasses import dataclass
+from typing import List, Optional, Tuple, Union
 
 from cuda import nvrtc
+from cuda.core.experimental._device import Device
 from cuda.core.experimental._module import ObjectCode
-from cuda.core.experimental._utils import handle_return
+from cuda.core.experimental._utils import _handle_boolean_option, handle_return
+
+
+@dataclass
+class ProgramOptions:
+    """Customizable :obj:`ProgramOptions` for NVRTC.
+
+    Attributes
+    ----------
+    gpu_architecture : str, optional
```

Comment:
For all options, not just this one: The compiler and the linker share a large number of overlapping options. Let's try to reuse the treatment we applied to Linker here. For example, in LinkerOptions this is simply `arch`.

Reply:
btw let's also unify the Sphinx formatting so that this renders correctly in doc build